### PR TITLE
Introduce "Integration" Subpage

### PR DESCRIPTION
**Pull request by christianwach**
Overview
----------------------------------------
Introduces an "Integration" page accessible via a sub-menu item for "CiviCRM" with links to lots of resources that will be of interest to those running CiviCRM under WordPress. Designed to be extensible via [the WordPress Meta Box framework](https://developer.wordpress.org/plugins/metadata/custom-meta-boxes/). Resources can be added or removed via PRs on this repo.

As originally suggested in the [Roadmap issue on Lab](https://lab.civicrm.org/dev/wordpress/-/issues/20).

Screenshot
----------------------------------------
Screen Shot 2020-11-27 at 13 45 28

### CODE REVIEW

**Comment by JoeMurray** [comment]
Very nice! Thanks for this work!

**Comment by christianwach** [comment]
@JoeMurray Thanks, it's taken a while but getting there. Next in the pipeline (as mentioned in the issue and once I've completed the logic) is the WordPress-specific settings screen to which I'll migrate some of the options in CiviCRM Admin Utilities.

Screen Shot 2020-11-27 at 13 52 56

File: assets/templates/metaboxes/metabox.repo.ext.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+<p><em><a href="<?php echo $extensions_url; ?>"><?php _e('Go to your CiviCRM Extensions Page', 'civicrm'); ?></a></em></p>
+
+<hr/>
+
+<p><?php _e('Here are some other places you can find them.', 'civicrm'); ?></p>
+
+<ul>
+  <li><em><a href="https://civicrm.org/extensions/wordpress" target="_blank"><?php _e('Search CiviCRM Website for "Extensions" that are compatible with WordPress', 'civicrm'); ?></a></em></li>
+  <li><em><a href="https://lab.civicrm.org/explore/projects?tag=wordpress" target="_blank"><?php _e('Search CiviCRM GitLab for "Extensions" that are compatible with WordPress', 'civicrm'); ?></a></em></li>
+  <li><em><a href="https://github.com/search?p=2&q=civicrm+extension&type=Repositories" target="_blank"><?php _e('Search GitHub for "CiviCRM Extensions"', 'civicrm'); ?></a></em></li>
```
Comment (braders) [inline assets/templates/metaboxes/metabox.repo.ext.php:32]: Presumably it is a mistake for this to be linking to page 2?

File: assets/templates/metaboxes/metabox.repo.git.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  <li><a href="https://lab.civicrm.org/extensions/wp-civicrm-mailer" target="_blank">CiviCRM Mailer for WordPress</a></li>
+  <li><a href="https://develop.tadpole.cc/plugins/civicrm-member-sync-single-role" target="_blank">CiviCRM Member Sync Single Role</a></li>
+  <li><a href="https://develop.tadpole.cc/plugins/civicrm-network-groups" target="_blank">CiviCRM Network Groups</a></li>
+  <li><a href="https://develop.tadpole.cc/plugins/civicrm-permissions-sync" target="_blank">CiviCRM Permissions Sync</a></li>
+  <li><a href="https://github.com/mecachisenros/content-views-civicrm" target="_blank">Content Views CiviCRM</a></li>
+  <li><a href="https://github.com/agileware/content-views-civicrm-data-processor" target="_blank">Content Views CiviCRM with CiviCRM Data Processor</a></li>
+  <li><a href="https://github.com/mecachisenros/woocommerce_civicrm" target="_blank">WooCommerce CiviCRM Integration</a></li>
+  <li><a href="https://github.com/agileware/wp-civicrm-ux" target="_blank">WordPress CiviCRM UX</a></li>
+  <li><a href="https://github.com/CiviCooP/wpcivi-shared" target="_blank">WPCivi Shared</a></li>
```
Comment (braders) [inline assets/templates/metaboxes/metabox.repo.git.php:46]: I'm not sure what the general CiviCRM approach to this is, but it feels like all external links on these metaboxes should contain `rel="noopener"`. See https://web.dev/external-anchors-use-rel-noopener/. Whilst these third-party sites are trusted, adding the attribute will protect CiviCRM site owners against potential hacks of lab.civicrm.org, develop.tadpole.cc etc

File: assets/templates/metaboxes/metabox.repo.ext.php (hunk 1 above)
Comment (christianwach) [reply assets/templates/metaboxes/metabox.repo.ext.php:32]: Nice catch @braders. Fixed in latest force push

File: assets/templates/metaboxes/metabox.repo.ext.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+</ul>
+
+<hr/>
+
+<p><?php _e('These are WordPress-only Extensions.', 'civicrm'); ?></p>
+
+<ul>
+  <li><a href="https://github.com/skyslasher/de.ergomation.wp-civi-mosaico" target="_blank">WordPress integration for CiviMail with Mosaico</a></li>
+  <li><a href="https://lab.civicrm.org/extensions/shoreditchwpworkarounds" target="_blank">Shoreditch WordPress Workarounds</a></li>
+</ul>
```
Comment (braders) [inline assets/templates/metaboxes/metabox.repo.ext.php:42]: What is the reasoning for these 2 extensions being listed separately from this list in `metabox.repo.git.php`?

File: assets/templates/metaboxes/metabox.repo.git.php (hunk 2 above)
Comment (christianwach) [reply assets/templates/metaboxes/metabox.repo.git.php]: @braders I've removed the `target="_blank"` attributes instead - they're not really necessary.

File: assets/templates/metaboxes/metabox.repo.ext.php (hunk 3 above)
Comment (christianwach) [reply assets/templates/metaboxes/metabox.repo.ext.php:42]: They're extensions not plugins.

Comment (braders) [reply assets/templates/metaboxes/metabox.repo.ext.php:42]: Ah yes, of course. Ignore me. This change as a whole looks good.

Comment (christianwach) [reply assets/templates/metaboxes/metabox.repo.ext.php:42]: Thanks for reviewing @braders

File: assets/templates/metaboxes/metabox.repo.git.php (hunk 2 above)
Comment (christianwach) [reply assets/templates/metaboxes/metabox.repo.git.php]: By that I mean it's not rocket science to open links in new tabs :-)

**Comment by christianwach** [comment]
One final force push to correct some comments and `@since` refs.

**Comment by christianwach** [comment]
Closed in favour of #231

**Pull request closed without merging**